### DISCUSSION TITLE
Babashka main function in the command line but not REPL

### DISCUSSION THREAD

**Discussion by oyarsa** [babashka/babashka#1565] [Q&A] [answered]
I'm just starting with Babashka, but something is bugging me: how can I ensure that a function (`-main` or whatever) is only run from the command line? I don't want it to run when I'm evaluating my file in the REPL.

My current approach is to wrap the top-level forms in a `run` function, comment its invocation, and only uncomment when I'm done developing and want to run it as a script. That's not very clean, though, so I wonder if there's something similar to Python's `if __name__ == '__main__'` to take care of that automatically.

I know Babashka supports running the `-main` function if you invoke it correctly, but for simple scripts, it involves changing the classpath to point to the current directory, and I don't get tab completion on the shell to pick the file.

**Comment by borkdude** [comment, marked as answer]
Hey @oyarsa !

We have this pattern for your use case, very similar to Python's pattern:

https://book.babashka.org/#main_file

**Reply by oyarsa** [reply]
I checked the book, but somehow I missed this. Thank you!
